src/lib/fees/model.ts

```diff
@@ -59,30 +66,57 @@ export const calculateFees: CalculateFeeBps = ({
   const midpointUsd = bn(FEE_CURVE_MIDPOINT_USD)
   const feeCurveSteepness = bn(FEE_CURVE_STEEPNESS_K)
   const isThorFreeEnabled = getConfig().REACT_APP_FEATURE_THOR_FREE_FEES
+  const isFoxWifHatEnabled = getConfig().REACT_APP_FEATURE_FOX_PAGE_FOX_WIF_HAT_SECTION
+
+  const isFoxWifHatCampaignActive =
+    new Date().getTime() >= FOX_WIF_HAT_CAMPAIGN_STARTING_TIME_MS &&
+    new Date().getTime() <= FOX_WIF_HAT_CAMPAIGN_ENDING_TIME_MS &&
+    isFoxWifHatEnabled
+
+  const currentFoxWifHatDiscountPercent = (() => {
+    if (!foxWifHatHeld || foxWifHatHeld?.lt(FOX_WIF_HAT_MINIMUM_AMOUNT_BASE_UNIT)) return bn(0)
+
+    const currentTime = new Date().getTime()
+    const totalCampaignDuration =
+      FOX_WIF_HAT_CAMPAIGN_ENDING_TIME_MS - FOX_WIF_HAT_CAMPAIGN_STARTING_TIME_MS
+    const timeElapsed = currentTime - FOX_WIF_HAT_CAMPAIGN_STARTING_TIME_MS
+    const remainingPercentage = bn(100).times(
+      bn(1).minus(bn(timeElapsed).div(totalCampaignDuration)),
+    )
+
+    return BigNumber.maximum(BigNumber.minimum(remainingPercentage, bn(100)), bn(0))
+  })()
 
   // trades below the fee threshold are free.
   const isFree = tradeAmountUsd.lt(noFeeThresholdUsd)
 
   const isThorFree =
     isThorFreeEnabled &&
-    thorHeld?.isGreaterThanOrEqualTo(THORSWAP_UNIT_THRESHOLD) &&
+    thorHeld?.gte(THORSWAP_UNIT_THRESHOLD) &&
     new Date().getUTCFullYear() < THORSWAP_MAXIMUM_YEAR_TRESHOLD
 
   // failure to fetch fox discount results in free trades.
-  const isFallbackFees = isSnapshotApiQueriesRejected
+  const isFallbackFees =
+    isSnapshotApiQueriesRejected &&
+    (!foxWifHatHeld ||
+      foxWifHatHeld?.lt(FOX_WIF_HAT_MINIMUM_AMOUNT_BASE_UNIT) ||
+      !isFoxWifHatCampaignActive)
```

Comment:
blocking: this doesn't look correct, see comment above on l.98 on why these are fallback free fees.

With this diff, trades (and not only, e.g LPs) will be free if If anyOf:
- FWF portfolio balance is undefined
- campaign inactive 
- FWF amount too low

is true, which we don't want? 

Suggestion: keep line as it was before. Held WFH is coming from portfolio balance and isn't an explicit snapshot query that is at high risk of being rejected, it's just good ol' portfolio business and should already be handled gracefully with `undefined` checks (no balance, though I think this is more of TS paranoia than anything else, *should* always be defined iirc) and actual 0 in case of 0 balance, there really isn't a failure case here.

Reply:
Welp, I guess technically speaking, this would add one more additional expression to the `isFallbackFees` so high-level *should* be safer (2 > 1)... just feels like there is no need to handle this explicitly and add complexity for something that should already be handled gracefully, WDYT?

Reply:
Removed in [a3d9766](https://github.com/shapeshift/web/pull/8663/commits/a3d9766fb8ffbb18fd5aad5c1665b80a083e62e1)